src/main/java/console/Parser.java

```diff
@@ -0,0 +1,59 @@
+package console;
+
+public class Parser {
+    private final String input;
+    private final Command command;
+    private final String additionalInput;
+    private final boolean isSingleWord;
+
+    public Parser(String input) {
+        this.input = input;
+
+        // Parse first word to get command
+
+        String firstWord = input;
+        int firstSpaceIndex = input.indexOf(" ");
+        if (firstSpaceIndex != -1) {
+            firstWord = input.substring(0, firstSpaceIndex);
+        }
+        this.command = Command.getEnum(firstWord);
+
+        // Parse additional input
+
+        int additionalInputIndex = command.toString().length() + 1;
+        boolean isNotDefaultCommand = command != Command.DEFAULT;
+        boolean isAdditionalInputExists = additionalInputIndex < input.length();
+        if (isNotDefaultCommand && isAdditionalInputExists) {
+            // if it is non-DEFAULT commands and there is additional input
+            this.additionalInput = input.substring(additionalInputIndex);
+        } else {
+            this.additionalInput = "";
+        }
+
+        this.isSingleWord = additionalInput.isEmpty();
+    }
+
+    public String getInput() {
+        return input;
+    }
+
+    public Command getCommand() {
+        return command;
+    }
+
+    public String getAdditionalInput() {
+        return additionalInput;
+    }
+
+    public boolean isSingleWord() {
```

Comment:
Maybe change the method name to have a verb at the start, such as checkIfSingleWord.